File: examples/spring-java/src/main/java/com/example/pets/PetController.java
package com.example.pets;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/pets")
public class PetController {

    @GetMapping("/{petId}")
    public Pet getPet(@PathVariable long petId) {
        return new Pet(petId, "Example pet");
    }

    public record Pet(long id, String name) {}
}

File: examples/spring-kotlin/src/main/kotlin/com/example/pets/PetController.kt
package com.example.pets

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/pets")
class PetController {

    @GetMapping("/{petId}")
    fun getPet(@PathVariable petId: Long): Pet = Pet(petId, "Example pet")
}

data class Pet(val id: Long, val name: String)
